### PR TITLE
Initial commit of XFS runtime statistics

### LINKED ISSUE #35: Parse XFS stats
Looks like there could be some useful information in here:

```
matt@servnerr-2:~$ cat /sys/fs/xfs/stats/stats
extent_alloc 92021 97163 92022 93325
abt 0 0 0 0
blk_map 1760270 187968 184039 92021 92022 2132277 0
bmbt 0 0 0 0
dir 184613 92021 92018 136144
trans 706 940035 0
ig 184619 58536 0 126083 0 33482 22
log 2871 112984 9 17314 739
push_ail 940745 0 132520 15426 0 3940 240 154770 0 40
xstrat 92021 0
rw 107225 93619
attr 4 0 0 0
icluster 8648 7824 135243
vnodes 92601 0 0 0 92018 92018 92018 0
buf 2654846 7122 2647761 3599 2 7085 0 10297 7085
abtb2 184089 1271462 13227 13193 0 0 0 0 0 0 0 0 0 0 2733130
abtc2 343698 2405723 171866 171832 0 0 0 0 0 0 0 0 0 0 21306938
bmbt2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
ibt2 341422 1352238 0 0 0 0 0 0 0 0 0 0 0 0 0
fibt2 0 0 0 0 0 0 0 0 0 0 0 0 0 0 0
qm 0 0 0 0 0 0 0 0
xpc 397979648 92452093 85848224
debug 0
```

Feel free to assign this to me. To be done for https://github.com/prometheus/node_exporter/issues/456.

### PR DESCRIPTION
**Pull request by mdlayher**
Parses XFS stats from `/proc/fs/xfs/stat`.  Appears these move to `/sys` in a later release, but a symlink remains, so I've kept it simple.

There are a decent number of counters that are undocumented so I've chosen to ignore those fields for now unless I can find some better resources to explain what they're actually for.

Fixes #35 .

### CODE REVIEW

File: xfs/parse.go [hunk 1]
```diff
@@ -0,0 +1,348 @@
+package xfs
```
Comment (grobie) [inline xfs/parse.go]: Missing license header.

File: xfs/parse_test.go [hunk 2]
```diff
@@ -0,0 +1,433 @@
+package xfs_test
```
Comment (grobie) [inline xfs/parse_test.go]: License header.

File: xfs/xfs.go [hunk 3]
```diff
@@ -0,0 +1,145 @@
+// Package xfs provides access to statistics exposed by the XFS filesystem.
```
Comment (grobie) [inline xfs/xfs.go]: Missing license header.

**Comment by mdlayher** [comment]
Fixed, @grobie . Also filed #37 because I noticed many files are missing license headers.

**Comment by mdlayher** [comment]
> Looks good. Wonder whether procfs is the right repository, given its being moved to /sys, but fine to keep it here for now.

I asked @SuperQ on IRC about this and he mentioned that some other code in here uses `/sys` too.  Worst case, it's easy enough to move out later.

**Comment by grobie** [comment]
That's not correct, everything in the procfs library is /proc related so
far. I guess he confused it with node_exporter.

Given the procfs package is marked as wip, I'm fine adding it for now.

**Pull request merged**